gce-scripts/README.md

```diff
@@ -0,0 +1,205 @@
+        *   which will make all the scripts executable.
+    *   `touch domains.csv`
+        *   Your domain list, one domain per line, with the input list ending in
+            `.csv`.
+        *   domains must have the schema stripped of them and no trailing '/',
+            such as:
+            *   domain.tld/path/to/page
+            *   domain.tld
+            *   subdomain.domain.tld
+            *   www.subdomain.domain.tld
+    *   `mkdir ~/pshtt_run/data_results/`
+    *   `mv ~/pshtt_run/combine_shards.py ~/pshtt_run/data_results`
+        *   Places combine_shards.py into data_results/.
+    *   `mkdir ~/pshtt_run/input_files/`
+
+4.  roots.pem
+
+    We want to use our own CA file when running pshtt. We use the mozilla root
+    store for this purpose. Follow instructions on this
+    [PR](https://github.com/agl/extract-nss-root-certs).
+
+5.  Updating ssh key
+
+    *   If your new ssh key is called "gce_pshtt_key", skip this step.
+    *   If you did not name your ***new*** ssh key gce_pshtt_key, then you will
+        need to go through and rename the gce_pshtt_key in all the .sh files to
+        whatever you named your key.
+    *   in vim, this is ":%s/gce_pshtt_key/yourkeynamehere/g <enter>".
+
+## How to run
+
+1.  `screen -S pshtt_running`
+2.  `cd ~/pshtt_run/`
+3.  `./run_all_scripts <input_file_name> <number_of_shards> <shard_name> >
+    log.out`
+    *   number of shards == number of hosts
+    *   each machine will contain a shard of the data to run.
+    *   This is the script that sets up all machines and puts all datafiles on
+        the machines for running.
+    *   `./run_all_scripts top-1m.nocommas.8.31.2017 100 alexa`
+    *   will produce 100 shards all starting with "alexa" in the input_files
+        dir.
+        *   ex. alexa000.csv
+    *   NOTE: you can ONLY create 999 shards. If you need more than 999 shards,
+        you will need to change the split_up_dataset.sh file.
+4.  exit screen `cntr+a+d`
+
+## During the run
+
+*   `./check_instances.sh`
+    *   will print the ip of each host, as well as FINISHED or NOT FINISHED.
+
+## After the run
+
+*   `./grab_and_combine_data.sh`
+
+    *   will grab all log and result data files, combine data files into one
+        large result file, and put these into data_results/.
+
+*   Delete your instance group. If you want to run data analysis, jump down to
+    the data analysis portion.
+
+# Running Pshtt on your local machine
+
+1.  Copy packages_to_install.sh and install the packages_to_install.sh.
+    *   `sudo ./packages_to_install.sh`
+2.  Clone pshtt.
+    *   `git clone https://github.com/dhs-ncats/pshtt.git`
+3.  Put roots.pem, running_script.sh, and your input file in the same dir as
+    pshtt.
+    *   Follow directions under Updating data files above on how to get a
+        roots.pem.
+    *   domains must have the schema stripped of them and no trailing '/', such
+        as:
+        *   domain.tld/path/to/page
```

Comment:
`domain.tld/path/to/page` isn't a supported thing to do with `pshtt`, and unfortunately right now will even screw up the base domain calculator.